### PR TITLE
Deprecate ChoiceFieldset and ChoiceInputField

### PR DESCRIPTION
**Pull request by mperrotti**

### CODE REVIEW

**Comment by mperrotti** [comment]
Duplicate of https://github.com/primer/react/pull/1891

File: .changeset/nervous-pets-sleep.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+</td>
+<td valign="top">
+
+```jsx
+import {FormGroup, Checkbox} from "@primer/react";
+
+// Multi-select
+<CheckboxGroup>
+  <CheckboxGroup.Label>Prefered Primer component interface</CheckboxGroup.Label>
```
Comment (rezrah) [inline .changeset/nervous-pets-sleep.md:55]: small typo

```suggestion
  <CheckboxGroup.Label>Preferred Primer component interface</CheckboxGroup.Label>
```

File: .changeset/nervous-pets-sleep.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+---
+'@primer/react': major
+---
+
+![image](https://user-images.githubusercontent.com/2313998/155632630-60822d7b-6053-480d-925d-cea4d3b04919.png)
```
Comment (rezrah) [inline .changeset/nervous-pets-sleep.md:5]: If we go with these banners in the change logs*, can we change this to a more descriptive caption here please? 

Also, wondering what you think about switching the labels and group headings to be more product-oriented in the banner, like they are in code snippets? 

* Not sure if we are doing banners, as other PRs don't include them. We'd need to backfill them.

File: .changeset/nervous-pets-sleep.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    <ChoiceFieldset.Item value="figma">Figma library</ChoiceFieldset.Item>
+    <ChoiceFieldset.Item value="css">Primer CSS</ChoiceFieldset.Item>
+    <ChoiceFieldset.Item value="react">Primer React components</ChoiceFieldset.Item>
+    <ChoiceFieldset.Item value="viewcomponents">Primer ViewComponents</ChoiceFieldset.Item>
+  </ChoiceFieldset.List>
+</ChoiceFieldset>
+
+// Single select
+<ChoiceFieldset>
+  <ChoiceFieldset.Legend>Prefered Primer component interface</ChoiceFieldset.Legend>
```
Comment (rezrah) [inline .changeset/nervous-pets-sleep.md:35]: FYI typo on preferred

File: .changeset/nervous-pets-sleep.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+![image](https://user-images.githubusercontent.com/2313998/155632630-60822d7b-6053-480d-925d-cea4d3b04919.png)
+
+The `CheckboxGroup` and `RadioGroup` components will be used to deprecate the `ChoiceFieldset` component. The deprecation of `ChoiceFieldset` also allows us to deprecate `ChoiceInputField`.
+
+`CheckboxGroup` and `RadioGroup` have the ability to render contextual content with your fieldset: labels, validation statuses, captions. It also handles the ARIA attributes that make the form controls accessible to assistive technology.
+
+<table>
+<tr>
+<th> Before </th> <th> After </th>
```
Comment (rezrah) [inline .changeset/nervous-pets-sleep.md:13]:
```suggestion
<th> Before (v34) </th> <th> After (v35)</th>
```

File: .changeset/nervous-pets-sleep.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+</tr>
+<tr>
+<td valign="top">
+
+```jsx
+import {ChoiceFieldset} from "@primer/react";
+
+// Multi-select
+<ChoiceFieldset>
+  <ChoiceFieldset.Legend>Prefered Primer component interface</ChoiceFieldset.Legend>
```
Comment (rezrah) [inline .changeset/nervous-pets-sleep.md:23]: The before / after is overflowing quite a lot.. is it possible to add more line breaks to the `before` snippet so it brings after more into view?

```suggestion
  <ChoiceFieldset.Legend>
    Prefered Primer component interface
  </ChoiceFieldset.Legend>
```

File: src/index.ts [hunk 6]
```diff
@@ -8,14 +8,6 @@ export type {ThemeProviderProps} from './ThemeProvider'
...
 export {default as Box} from './Box'
 export type {BoxProps} from './Box'
-export {PageLayout} from './PageLayout'
-export type {
-  PageLayoutProps,
-  PageLayoutHeaderProps,
-  PageLayoutContentProps,
-  PageLayoutPaneProps,
-  PageLayoutFooterProps
-} from './PageLayout'
```
Comment (siddharthkp) [inline src/index.ts:18]: This seems unintentional?

**Comment by siddharthkp** [comment]
Using this checklist for review:

### User facing

- [x] Add changelog

API:
  - [x] Move from src/index to src/deprecated
  - [x] Add @deprecated JSDoc annotation

Docs:
  - [x] Update status to deprecated in docs
  - [ ] Add 'deprecated' to doc code examples (for ChoiceInputField)
  - [x] Add deprecation notice in docs
  - [ ] Move to deprecated section in docs/nav.yml
  - [ ] Move to deprecated folder in docs/
  

### Internal maintenance

- NA Update import path in related components

Tests
- [ ] Move to tests/deprecated/
- [ ] Update import path in tests
- [ ] Update test snapshots

Storybook:
- NA Update title in story meta "Deprecated components"
- NA Move to stories/deprecated/
- NA Update import path in stories

### Checks
- [ ] Docs working x (ChoiceInputField docs broken)
- [ ] Tests passing x
- [x] Storybook working

**Review by siddharthkp — CHANGES_REQUESTED** [review]
Using this checklist for review:

### Checks
- [ ] Docs working x (ChoiceInputField docs broken)
- [ ] Tests passing x
- [x] Storybook working

### User facing (should prioritise for release)

- [x] Add changelog

API:
  - [x] Move from src/index to src/deprecated
  - [x] Add `@deprecated` JSDoc annotation

Docs:
  - [x] Update status to deprecated in docs
  - [x] Add 'deprecated' to doc code examples (for ChoiceInputField)
  - [x] Add deprecation notice in docs
  - [ ] Move to deprecated section in docs/nav.yml
  - [ ] Move to deprecated folder in docs/
  

### Internal maintenance (not blocking, can tackle in a new PR)

- NA Update import path in related components

Tests
- [x] Move to tests/deprecated/
- [x] Update import path in tests
- [x] Update test snapshots

Storybook:
- NA Update title in story meta "Deprecated components"
- NA Move to stories/deprecated/
- NA Update import path in stories

File: .changeset/nervous-pets-sleep.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
+---
+'@primer/react': major
+---
+
+The `CheckboxGroup` and `RadioGroup` components will be used to deprecate the `ChoiceFieldset` component. The deprecation of `ChoiceFieldset` also allows us to deprecate `ChoiceInputField`.
```
Comment (rezrah) [inline .changeset/nervous-pets-sleep.md:5]: If it's a quick one, can we add the heading in? Happy to do it in the final changelog too, so no biggie

```
### ChoiceFieldset and ChoiceFieldset
```

File: src/deprecated/ChoiceInputField.tsx [hunk 8]
```diff
@@ -25,6 +25,9 @@ const getInputToRender = (inputType: 'radio' | 'checkbox', children?: React.Reac
   )
 }
 
+/**
+ * @deprecated Use `FormControl` instead. See https://primer.style/react/FormControl for more info
```
Comment (rezrah) [inline src/deprecated/ChoiceInputField.tsx:29]: Have you verified this annotation works? I think it'll need to be applied at the bottom of the file in the object assignment instead.

**Review by rezrah — APPROVED** [review]
looks great.. just one observation here: https://github.com/primer/react/pull/1900/files#r816806465

File: src/deprecated/ChoiceFieldset/ChoiceFieldset.tsx [hunk 9]
```diff
@@ -57,7 +57,7 @@ const {Slots, Slot} = createSlots(['Description', 'ChoiceList', 'Legend', 'Valid
 export {Slot}
 
 /**
- * @deprecated Use `CheckboxGroup` or `RadioGroup` instead.
+ * @deprecated Use `CheckboxGroup` or `RadioGroup` instead. See https://primer.style/react/CheckboxGroup and https://primer.style/react/RadioGroup for more info
```
Comment (rezrah) [inline src/deprecated/ChoiceFieldset/ChoiceFieldset.tsx:60]: Similar feedback to the other annotation, I think this needs to be applied here:

```js
/**
  * @deprecated Use `CheckboxGroup` or `RadioGroup` instead. See https://primer.style/react/CheckboxGroup and https://primer.style/react/RadioGroup for more info
  */
export default Object.assign(ChoiceFieldset, {

```

**Review by rezrah — APPROVED** [review]

**Pull request merged**